www-client/firefox: add jpegxl support

l29ah opened the pull request:
Bug: https://bugs.gentoo.org/928126


<!-- Please put the pull request description above -->

---

Please check all the boxes that apply:

- [x] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [x] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [x] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [x] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

juippis commented on www-client/firefox/firefox-137.0.2.ebuild:
```diff
@@ -600,6 +600,7 @@ src_prepare() {
 
 	eapply "${WORKDIR}/firefox-patches"
 	use loong && eapply "${WORKDIR}/firefox-loong-patches"
+	use jpegxl && eapply "${FILESDIR}/firefox-133.0.3-jxl-enable.patch"
```
No, the patch needs to be always applied and `--disable-jxl` used instead based on the use flag.

juippis reviewed (commented):
To my surprise, they still keep updating the bundled jxl even when they're rejected the idea of using it. Does it still actually work?

I know they updated their policy to implement jxl if it gets rewritten in rust - but nothing seems to be happening on that area.

So I'm kind of 50-50 on this. The bundled lib is updated, but they don't seem keen on keeping it. Anyway, could target 138.0 release which is out next week if jxl still works just fine in the browser?

l29ah commented:
Fixed and updated, tested with https://tinystash.undef.im/il/2wKWQhj8t5bHpCk9q3BqxKbgoojxJvexZNhQ4PNJhQ4AUydUmzdBfRnQQDTgXqdfU3oc3Dt5qDPfX5Sv79ygjQrT.jxl
